### ISSUE TITLE
Control over downloaded/extracted files naming

### ISSUE THREAD

**Issue by remydubois** [tensorflow/datasets#991]
Hello,

I often encounter the situation where I want to 'explore' or inspect the raw data (which is downloaded and extracted) before treating it and preprocessing it into TFRecord files.

However, the naming of those folders (resulting from the extraction) (based on a hashing of the checksum and the url if I understood [this line](https://github.com/tensorflow/datasets/blob/800ce69fe928670694f1a7f8715a2e2379683795/tensorflow_datasets/core/download/resource.py#L169)) makes it hard because names are non-human friendly.

I was wondering whether controlling this behavior would be possible,

Thanks

**Comment by Suhasnama** [comment]
@remydubois , @Conchylicultor , I would like to work on this issue. Could you please help me get started with this issue.
Thank you!!

**Comment by Conchylicultor** [comment]
I agree the current names could be made more user friendly, but we rely on the hashing mechanism to guarantee that datasets are reproducible and check the integrity of the files.
This would require some though from our side to satisfy both requirements.
@pierrot0, do you have some though on this ?

**Comment by pierrot0** [comment]
One possible way would be to use symlinks.
Symlinks aren't supported on all file systems, so the canonical dir name could stay the same, and if symlink is possible, symlink from ${dataset_name}_%{fname} to downloaded file.
This way, if two datasets use the same file, file is accessible through both names.

Would that work for you Remy?

**Comment by remydubois** [comment]
Thank you for your answers.

Agreed, that would be a good way around to expose more human friendly raw datasets.

I do not know how this could be done while altering as less as possible the current API. 
Perhaps the ´tfds.download.Resource´  object could accept a ´symlink´ argument (?).

Explanations: my current workflow is such that I build a table summarizing my dataset with a bunch of metadata about each file to download (like a meaningful name), including its URL. With the proposed ´Resource ´ object, I could directly build a meaningful symlink from the metadata for each file to download, and feed the set of Resources to the ´downloadmanager´

**Comment by Suhasnama** [comment]
Hello @Conchylicultor , @pierrot0  could you please help get started with this issue!!
